### PR TITLE
Update ShareLink sample

### LINKED ISSUE #21543: [QUERY] KeyVault: Sharelink sample can not be built in vs code
**Environment:**
Sharelink 

I download sharelink.zip and extract it, but build in vs code, error is:
...\ShareLink\Properties\AssemblyInfo.cs(4,23): error CS0234: The type or namespace name 'CodeGenSuppressTypeAttributeAttribute' does not exist in the namespace 'Azure.Core' (are you missing an assembly reference?)

can anyone give me some idea? thanks

### PR DESCRIPTION
**Pull request by heaths**
Fixes #21543

### CODE REVIEW

**Comment by heaths** [comment]
I'm pondering just splatting all the files in this sample we need and keep it as a static copy to avoid updates breaking the sample. An updated sample went out that doesn't build when downloaded from Samples Browser because of some refactoring. Some files weren't added to the autorest.csharp package, which I also had an older version for in the .csproj to make sure it builds.

This should fix the problem for now, but we may need to think longer-term about samples.

**Comment by christothes** [comment]
Is there a reason why we couldn't add this sln to the paths that get build in CI?

**Comment by pakrym** [comment]
Yeah, we (Heath) have been fixing this sample way too many times by now. We need something to run it in CI.

File: sdk/keyvault/samples/sharelink/ShareLink.csproj [hunk 1]
```diff
@@ -34,7 +34,7 @@
     <PackageReference Update="Azure.Core" Version="1.14.0" />
     <PackageReference Update="Azure.Identity" Version="1.4.0" />
     <PackageReference Update="Azure.Security.KeyVault.Secrets" Version="4.1.1" />
-    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20210510.2" PrivateAssets="All" />
+    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20210602.1" PrivateAssets="All" />
```
Comment (pakrym) [inline sdk/keyvault/samples/sharelink/ShareLink.csproj:37]: This package is not available on nuget do we expect customers to reference our tooling feed?

Comment (heaths) [reply sdk/keyvault/samples/sharelink/ShareLink.csproj:37]: It's available in our public feed and I copy the nuget.config from the repo. If we were to make this just a static copy of all the files, that would eliminate that problem as well.

**Comment by heaths** [comment]
@christothes it does actually build when sdk/keyvault is built (or the core pipeline). The problem is that it has all the advantages of the various eng/* targets and props, which it doesn't when built from the Samples Browser. It actually can build in three modes. I added a CopySource target which I originally did to try to simulate Samples Browser ingestion a little. It's not exactly the same, but does move all the source out of our repo (depending on your `/p:Destination` value) so I can build and see if it works successfully. It's even called out in the README.

I do think we need to continue discussing how to do samples easily, reliably, and with some benefits (e.g. build and code analyzers to make sure our samples are good and consistent), but without inheriting all the other free stuff that won't work outside the repo.

**Comment by heaths** [comment]
I got to thinking that if I delete the Shared\Core and Shared\AutoRest directories, copied the AuthenticationChallengeParser.cs file, and the nuget.config I could better simulate what Samples Browser would ZIP up, which is how I found I needed to set another property in that last commit (it's set automatically when built inside our repo).

File: sdk/keyvault/samples/sharelink/ShareLink.csproj (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/samples/sharelink/ShareLink.csproj:38]: I better simulated the ZIP and did find a missing prop to actually make this really work. See my [comment](https://github.com/Azure/azure-sdk-for-net/pull/21598#issuecomment-854996755) below for details.

**Review by pakrym — APPROVED** [review]

**Pull request merged**